### ISSUE TITLE
Aragog output

### ISSUE THREAD

**Issue by lsoucasse** [ExPlanetology/aragog#15]
The output module of Aragog already features many field quantities like temperature, melt fraction, pressure... But we would like it to output some other scalar quantities, namely the:
- [x] Surface temperature
- [x] Surface heat flux (point to the convective heat flux output)
- [x] Global melt fraction (melt fraction volume average)
- [x] Liquid, solid and total mantle mass (computed from pressure field and equation of state?)
- [x] Depth of rheological front (from user defined threshold)

**Comment by lsoucasse** [comment]
It seems most of the output functions are not working because it uses a data attribute that does no longer exist. I guess we have to replace it by the evaluator attribute, right @djbower ?

**Comment by djbower** [comment]
It might be more straightforward to use the `State` class to avoid duplication. Basically, the `State` class should self-consistently update a bunch of quantities using the `update` method, which takes a temperature and a time as an argument.  As I mentioned on the call today, this updates exclusively the functions required to do the integration, so there might be a few more output options that are desired.  But I would probably start by using `State`, and then extending the class (or wrapping it with another function) to add more output options.  Also note that `_basic` and `_staggered` refer to quantities on the basic and staggered mesh, respectively.  Another option would be to add more methods to the State class to compute the output options, even though these methods are not called to update the state for the purposes of integration.

Part of my motivation for the `State` class was to avoid multiple function calls, but the trade-off is the dangers of dealing with a state that must be self-consistently updated.

So in short, maybe it's simplest to leverage the `State` class to provide a more concise mechanism for output?

**Comment by djbower** [comment]
On this note, my usage of class `properties` versus class `methods` was a bit ad-hoc.  This could probably be cleaned up at the same time.

**Comment by djbower** [comment]
In fact, the melt fraction should be computed for all pressures in the `State` class anyway, since it is required for a couple of additional heat fluxes that are not yet implemented.  The melt fraction is defined to be unity above the liquidus and 0 below the solidus.  In the mixed phase region (between the liquidus and the solidus) it is computed by the Lever rule (T-Tsol)/(Tliq-Tsol).  Tsol and Tliq should be evaluated from the property lookup for each pressure/radius.

**Comment by lsoucasse** [comment]
I have implemented a function that gets the mantle mass from the Adams-Williamson EOS. We can derive liquid/solid counterparts using the global melt fraction. We can revise this later on to make it consistent with phase densities, see [here](https://github.com/ExPlanetology/aragog/issues/16#issuecomment-2416172865).

**Comment by timlichtenberg** [comment]
Hi Laurent, awesome! Can you have @planetmariana review and shortly discuss this with her? This would be a good start to look into the functioning of aragog to add further functions that Mariana will need to implement for her research projects.

**Comment by lsoucasse** [comment]
> Hi Laurent, awesome! Can you have @planetmariana review and shortly discuss this with her? This would be a good start to look into the functioning of aragog to add further functions that Mariana will need to implement for her research projects.

Sure! @planetmariana, I sent you a message on Discord.

**Comment by nichollsh** [comment]
I've made a related issue for writing data here: https://github.com/ExPlanetology/aragog/issues/19
